### PR TITLE
[silabs] Add silabs specific CLI

### PR DESCRIPTION
**Pull request by jepenven-silabs**
Add support for Silabs specific CLI.

### CODE REVIEW

File: examples/platform/silabs/efr32/uart.cpp [hunk 1]
```diff
@@ -397,8 +397,12 @@ int16_t uartLogWrite(const char * log, uint16_t length)
 
     UartTxStruct_t workBuffer;
     memcpy(workBuffer.data, log, length);
-    memcpy(workBuffer.data + length, "\r\n", 2);
-    workBuffer.length = length + 2;
+    workBuffer.length = length;
+    if (newLine)
+    {
+        memcpy(workBuffer.data + length, "\r\n", 2);
```
Comment (rcasallas-silabs) [inline examples/platform/silabs/efr32/uart.cpp:403]:
```suggestion
workBuffer.data[length++] = '\r';
workBuffer.data[length++] = '\n';
```

**Review by rcasallas-silabs — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

File: examples/platform/silabs/efr32/uart.cpp (hunk 1 above)
Comment (jepenven-silabs) [reply examples/platform/silabs/efr32/uart.cpp]: not needed anymore

File: examples/platform/silabs/matter_shell.cpp [hunk 2]
```diff
@@ -16,11 +16,19 @@
...
 #include <FreeRTOS.h>
 #include <lib/core/CHIPCore.h>
 #include <lib/shell/Engine.h>
 #include <task.h>
+#ifdef SL_CATALOG_CLI_PRESENT
+#include "sl_cli.h"
+#include "sl_cli_config.h"
+#include "sli_cli_io.h"
+static char sSilabsTXCLI[SL_CLI_OUTPUT_BUFFER_SIZE];
+static constexpr uint16_t sSilabsTXCLILenght = SL_CLI_OUTPUT_BUFFER_SIZE;
```
[the remark this answers is not in the document]
Comment (jepenven-silabs) [reply examples/platform/silabs/matter_shell.cpp]: outdated

File: examples/platform/silabs/matter_shell.cpp [hunk 3]
```diff
@@ -69,6 +77,82 @@ void WaitForShellActivity()
     ulTaskNotifyTake(pdTRUE, portMAX_DELAY);
 }
 
+#ifdef SL_CATALOG_CLI_PRESENT
+
+CHIP_ERROR cmd_silabs_dispatch(int argc, char ** argv)
```
[the remark this answers is not in the document]
Comment (jepenven-silabs) [reply examples/platform/silabs/matter_shell.cpp]: fixed

File: examples/platform/silabs/uart.h [hunk 4]
```diff
@@ -26,7 +26,7 @@ extern "C" {
 
 void uartConsoleInit(void);
 int16_t uartConsoleWrite(const char * Buf, uint16_t BufLength);
-int16_t uartLogWrite(const char * log, uint16_t length);
+int16_t uartLogWrite(const char * log, uint16_t length, bool newLine=true);
```
[the remark this answers is not in the document]
Comment (jepenven-silabs) [reply examples/platform/silabs/uart.h]: outdated

**Review by jmartinez-silabs — APPROVED** [review]

**Pull request merged**